doc/api/tls.md

```diff
@@ -583,6 +583,13 @@ if called on a server socket. The supported types are `'DH'` and `'ECDH'`. The
 
 For Example: `{ type: 'ECDH', name: 'prime256v1', size: 256 }`
 
+### tlsSocket.getFinished()
+
+* Returns: {undefined|Buffer} The latest `Finished` message that was sent
+to the socket, or `undefined` if no `Finished` message has been sent yet.
+
+Corresponds to the `SSL_get_finished` routine in OpenSSL.
```

Comment:
What are the actual about the contents of these buffers? How does the peer generate them? Should Node provide support for that too?

Reply:
I added `as part of a SSL/TLS handshake` phrase to the documentation in order to emphasize that the `Finished` messages are internal to OpenSSL implementation and are not provided by application.

Reply:
@codedot If these are internal to OpenSSL, how is the returned buffer meaningful to the caller?

Reply:
An article called [*Why is it unlikely to have a complete and alternative implementation of Ripple?*][1] describes in detail how the `Finished` messages are used to generate `Session-Signature` in `rippled` using the fact that both `Finished` messages are available on both sides of an SSL/TLS socket. Also, see ripple/rippled#2413 and [the comment][2] by @JoelKatz.

However, I would not suggest including this motivation into Node.js documentation because I do not think it would be wise to endorse such a questionable method.

[1]: https://medium.com/@alexberegszaszi/why-is-it-unlikely-to-have-a-complete-alternative-implementation-of-ripple-513c8f258238
[2]: https://github.com/ripple/rippled/issues/2413#issuecomment-368994321

Reply:
@codedot All of these only resources only describe *that* these functions are necessary to implement that protocol, but make no statements about what guarantees can be made about the *returned contents* of these buffers.

I realize that the OpenSSL documentation and source code don’t provide much more information either, but since apparently people found a use case for these functions, there must be sensible statements that we could make here in the documentation.

Reply:
@addaleax The contents of the `Finished` messages, their length, and even their presence are not guaranteed at all and depend on OpenSSL's internals and the particular SSL/TLS version used for a session.

Unfortunately, I cannot propose any better way of describing `tlsSocket.getFinished()` and `tlsSocket.getPeerFinished()` than that they just correspond to `SSL_get_finished()` and `SSL_get_peer_finished()` from `<openssl/ssl.h>`, respectively.

Reply:
> The contents of the `Finished` messages, their length, and even their presence are not guaranteed at all and depend on OpenSSL's internals and the particular SSL/TLS version used for a session.

That sounds like something we should document as well, I think? Do you know more about what makes these messages appear/not appear?

Also, maybe @indutny can help with describing what these functions do, since he approved this PR?

Reply:
@addaleax Well, I think Node.js should not provide more documentation than OpenSSL itself does. The modern versions do not include any `man(1)` pages for them like they used to. However, even [the old documentation][1] was not descriptive.

[1]: http://nixdoc.net/man-pages/Tru64/SSL_get_finished.3.html

Reply:
> Well, I think Node.js should not provide more documentation than OpenSSL itself does.

I’d agree, but the place to start fixing this would be the OpenSSL documentation then ;)

I’m just really not a fan of having documentation that doesn’t really describe what something does…

Reply:
> I’d agree, but the place to start fixing this would be the OpenSSL documentation then ;)

I'm afraid that's the best I can do :(

Reply:
@codedot I guess there’s no harm in opening an issue: https://github.com/openssl/openssl/issues/5509

Reply:
@addaleax Thank you very much for asking OpenSSL team for help. Documentation updated, please review.